### PR TITLE
JS: only set the file in the diagnostics message if the file is within the source root

### PR DESCRIPTION
**Pull request by erik-krogh**
This should fix https://github.com/github/codeql/issues/12734

I wasn't able to replicate locally, but the stack-trace in the issue provided enough information. 

It was this change that caused the null-pointer: https://github.com/github/codeql/pull/12113#discussion_r1126379069

### CODE REVIEW

**Review by aibaars — APPROVED** [review]
LGTM

**Review by aibaars — CHANGES_REQUESTED** [review]
Perhaps this should be for the rc/3.9 branch

File: javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java [hunk 1]
```diff
@@ -1239,12 +1239,11 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
         String msg = "A parse error occurred: " + StringUtil.quoteWithBackticks(err.getMessage().trim())
             + ". Check the syntax of the file. If the file is invalid, correct the error or [exclude](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning) the file from analysis.";
         // file, relative to the source root
-        String relativeFilePath = null;
+        DiagnosticLocation.Builder builder = DiagnosticLocation.builder();
         if (file.startsWith(LGTM_SRC)) {
-          relativeFilePath = file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString();
+          builder = builder.setFile(file.subpath(LGTM_SRC.getNameCount(), file.getNameCount()).toString());
```
Comment (henrymercer) [inline javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java:1244]: Should we skip diagnostics, or perhaps just diagnostic locations, for files outside the source root?  I'm not sure how useful it is to have a location with a range but without a file path.

**Comment by erik-krogh** [comment]
Sorry for the ping, bad change of branch. 
I'll reopen another PR.

**Pull request closed without merging**